components/pixels/snapshot.rs

```diff
@@ -180,31 +195,35 @@ impl Snapshot<SnapshotData> {
         }
     }
 
-    pub fn get_rect(&self, rect: Rect<u32>) -> Self {
-        let data = rgba8_get_rect(self.as_raw_bytes(), self.size(), rect).to_vec();
-        Self::from_vec(rect.size, self.format, self.alpha_mode, data)
-    }
-
-    // TODO: https://github.com/servo/servo/issues/36594
-    /*
-    /// # Safety
-    ///
-    /// This is safe if data is owned by this process only
-    /// (ownership is transferred on send)
-    pub unsafe fn from_shared_memory(
+    pub fn from_shared_memory(
         size: Size2D<u32>,
-        format: PixelFormat,
-        alpha_mode: AlphaMode,
-        ism: IpcSharedMemory,
+        format: SnapshotPixelFormat,
+        alpha_mode: SnapshotAlphaMode,
+        data: Arc<IpcSharedMemory>,
+        byte_range_bounds: impl RangeBounds<usize>,
     ) -> Self {
+        let range_start = match byte_range_bounds.start_bound() {
+            Bound::Included(bound) => *bound,
+            Bound::Excluded(bound) => *bound + 1,
+            Bound::Unbounded => 0,
+        };
+        let range_end = match byte_range_bounds.end_bound() {
+            Bound::Included(bound) => *bound,
+            Bound::Excluded(bound) => *bound - 1,
+            Bound::Unbounded => 0,
```

Comment:
This is wrong. When unbounded we should use data.len() and because the range below has end excluded we can keep pass-through excluded:


```suggestion
        let range_end = match byte_range_bounds.end_bound() {
            Bound::Included(bound) => *bound - 1,
            Bound::Excluded(bound) => *bound,
            Bound::Unbounded => data.len(),
```

Reply:
I've fixed the Unbounded, case but I think that the inclusive bound ( `0..=3`) should be `0,1,2,3` and the exclusive bound (`0..3`) should be `0,1,2` (ie bound - 1).